tests/foreman/data/certs.sh

```diff
@@ -0,0 +1,69 @@
+#!/bin/bash
+
+CERTS_DIR=certs
+
+THIRDPARTY_CA_CERT_NAME=ca-thirdparty
+if [[ ! -f "$CERTS_DIR/$THIRDPARTY_CA_CERT_NAME.key" || ! -f "$CERTS_DIR/$THIRDPARTY_CA_CERT_NAME.crt" ]]; then
+  echo "Generating CA"
+  openssl genrsa -out $CERTS_DIR/$THIRDPARTY_CA_CERT_NAME.key 2048
+  openssl req -x509 -new -nodes -key $CERTS_DIR/$THIRDPARTY_CA_CERT_NAME.key -sha256 -days 3650 -out $CERTS_DIR/$THIRDPARTY_CA_CERT_NAME.crt -subj "/CN=Thirdparty CA"
+else
+  echo "Thirdparty CA certificate exists. Skipping."
+fi
+
+CA_CERT_NAME=ca
+if [[ ! -f "$CERTS_DIR/$CA_CERT_NAME.key" || ! -f "$CERTS_DIR/$CA_CERT_NAME.crt" ]]; then
+  echo "Generating CA"
+  openssl genrsa -out $CERTS_DIR/$CA_CERT_NAME.key 2048
+  openssl req -x509 -new -nodes -key $CERTS_DIR/$CA_CERT_NAME.key -sha256 -days 3650 -out $CERTS_DIR/$CA_CERT_NAME.crt -subj "/CN=Test Self-Signed CA"
+else
+  echo "CA certificate exists. Skipping."
+fi
+
+CA_BUNDLE=ca-bundle
+if [[ ! -f "$CERTS_DIR/$CA_BUNDLE.crt" ]]; then
+  echo "Generating CA bundle"
+  cat $CERTS_DIR/$THIRDPARTY_CA_CERT_NAME.crt $CERTS_DIR/$CA_CERT_NAME.crt > $CERTS_DIR/$CA_BUNDLE.crt
+else
+  echo "CA certificate bundle exists. Skipping."
+fi
+
+CERT_NAME=foreman.example.com
```

Comment:
We can eliminate these duplicate lines by creating a common function.
```
function cert_check(){
   CERT_NAME = $1
   if [[ ! -f "$CERTS_DIR/${CERT_NAME}.key" || ! -f "$CERTS_DIR/$CERT_NAME.crt" ]]; then
        echo "Generating server certificate"
        openssl genrsa -out $CERTS_DIR/$CERT_NAME.key 2048
        openssl req -new -key $CERTS_DIR/$CERT_NAME.key -out $CERTS_DIR/$CERT_NAME.csr -subj "/CN=foreman.example.com"
        openssl x509 -req -in $CERTS_DIR/$CERT_NAME.csr -CA $CERTS_DIR/$CA_CERT_NAME.crt -CAkey $CERTS_DIR/$CA_CERT_NAME.key -CAcreateserial -out $CERTS_DIR/$CERT_NAME.crt -days 3650 -sha256 -extfile extensions.txt -extensions extensions
    else
        echo "Server certificate exists. Skipping."
   fi
}

for cert_name in foreman.example.com  invalid wildcard shortname ; do
    cert_check $cert_name
done
```

Reply:
We also need to check the -subj "/CN=foreman.example.com" which will vary in multiple cases. just not thinking over-complicate this multiple for loops

Reply:
yes, you are right, you can achieve it in this simpler way.
```
function cert_check(){
   CERT_NAME=$1   
   SUBJ=$2
   if [[ ! -f "$CERTS_DIR/${CERT_NAME}.key" || ! -f "$CERTS_DIR/$CERT_NAME.crt" ]]; then
        echo "Generating server certificate"
        openssl genrsa -out $CERTS_DIR/$CERT_NAME.key 2048
        openssl req -new -key $CERTS_DIR/$CERT_NAME.key -out $CERTS_DIR/$CERT_NAME.csr -subj $SUBJ
        openssl x509 -req -in $CERTS_DIR/$CERT_NAME.csr -CA $CERTS_DIR/$CA_CERT_NAME.crt -CAkey $CERTS_DIR/$CA_CERT_NAME.key -CAcreateserial -out $CERTS_DIR/$CERT_NAME.crt -days 3650 -sha256 -extfile extensions.txt -extensions extensions
    else
        echo "Server certificate exists. Skipping."
   fi
}
cert_check "foreman.example.com" "/CN=foreman.example.com"
cert_check "invalid" "/CN=*.example.com"
cert_check "wildcard" "/CN=foreman"
cert_check "shortname" "/CN=Thirdparty CA"


```

Reply:
okay @devendra104 thanks for the efforts let me try that